mongodbatlas/resource_mongodbatlas_advanced_cluster.go

```diff
@@ -1124,6 +1125,36 @@ func resourceClusterAdvancedRefreshFunc(ctx context.Context, name, projectID str
 	}
 }
 
+func resourceClusterListAdvancedRefreshFunc(ctx context.Context, projectID string, client *matlas.Client) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		clusters, resp, err := client.AdvancedClusters.List(ctx, projectID, nil)
+
+		if err != nil && strings.Contains(err.Error(), "reset by peer") {
+			return nil, "REPEATING", nil
+		}
+
+		if err != nil && clusters == nil && resp == nil {
+			return nil, "", err
+		} else if err != nil {
```

Comment:
```suggestion
		} 
	 if err != nil {
```

Reply:
I applied these suggestions in another commit, alongside similar changes. Thanks for the feedback!